radix-engine-interface/src/api/types/actor.rs

```diff
@@ -6,23 +6,58 @@ use crate::model::*;
 use crate::scrypto;
 use crate::Describe;
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum PackageIdentifier {
+    Scrypto(PackageAddress),
+    Native(NativePackage),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum NativePackage {
+    Auth,
+    Component,
+    Package,
+    Metadata,
+    EpochManager,
+    Resource,
+    Clock,
+    Logger,
+    TransactionRuntime,
+    TransactionProcessor,
+}
```

Comment:
They seem to be blueprint equivalent rather than package, no?

Reply:
Should be package, for example, `Resource` maintains all Vaults/Bucket/Proof logic.

Reply:
Cool, fine with me for now. I'm just a bit concerned if we really need divide native nodes into multiple packages. It seems to me that we may need a "blueprint" virtual badge, on top of package badge, to provide finer granularity in access control.

Reply:
Yup, good point but I'm assuming we won't ever need a "blueprint" badge. The main value of such a badge is to not allow outside code apart from the "owner" of the code to be able to make certain calls. For internal protection, one could just use rust separation of files to protect the code owner from himself.